pkg/cstor/newpool/v1alpha3/build.go

```diff
@@ -0,0 +1,266 @@
+		)
+		return b
+	}
+	b.CSP.Object.Annotations = make(map[string]string)
+	for key, value := range annotations {
+		b.CSP.Object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithAnnotations appends or overwrites existing Annotations
+// values of CSP with provided arguments
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing annotations"),
+		)
+		return b
+	}
+	if b.CSP.Object.Annotations == nil {
+		return b.WithAnnotationsNew(annotations)
+	}
+	for key, value := range annotations {
+		b.CSP.Object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithLabelsNew sets the Labels field of CSP with provided arguments
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing labels"),
+		)
+		return b
+	}
+	b.CSP.Object.Labels = make(map[string]string)
+	for key, value := range labels {
+		b.CSP.Object.Labels[key] = value
+	}
+	return b
+}
+
+// WithLabels appends or overwrites existing Labels
+// values of CSP with provided arguments
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing labels"),
+		)
+		return b
+	}
+	if b.CSP.Object.Labels == nil {
+		return b.WithLabelsNew(labels)
+	}
+	for key, value := range labels {
+		b.CSP.Object.Labels[key] = value
+	}
+	return b
+}
+
+// WithNodeSelector sets the node selector field of CSP with provided argument.
+func (b *Builder) WithNodeSelector(nodeSelector map[string]string) *Builder {
+	if len(nodeSelector) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing nodeSelector"),
+		)
+		return b
+	}
+	b.CSP.Object.Spec.NodeSelector = nodeSelector
+	return b
+}
+
+// WithNodeName sets the HostName field of CSP with the provided argument.
+func (b *Builder) WithNodeName(nodeName string) *Builder {
+	if len(nodeName) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing node name"),
+		)
+		return b
+	}
+	b.CSP.Object.Spec.HostName = nodeName
+	return b
+}
+
+// WithPoolConfig sets the pool config field of the CSP with the provided config.
+func (b *Builder) WithPoolConfig(poolConfig *apis.PoolConfig) *Builder {
+	if poolConfig == nil {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing poolConfig"),
+		)
+		return b
+	}
+	b.CSP.Object.Spec.PoolConfig = *poolConfig
+	return b
+}
+
+// WithRaidGroups sets the raid group field of the CSP with the provided raid groups.
+func (b *Builder) WithRaidGroups(raidGroup []apis.RaidGroup) *Builder {
+	if len(raidGroup) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build CSP object: missing raidGroups"),
+		)
+		return b
+	}
+	b.CSP.Object.Spec.RaidGroup = raidGroup
```

Comment:
can we have some mechanism to copy raidGroup to csp instead of directly assigning address .
If we are getting raidGroup argument as pass by value then we can ignore this comment.